### DISCUSSION TITLE
DayPickerInput doesn't close after changing month when using a custom component

### DISCUSSION THREAD

**Discussion by florianchevallier** [gpbl/react-day-picker#1158] [Support]
**Describe the bug**
When using a DayPickerInput, if you use a custom component like this

```js
<DayPickerInput
        placeholder="DD/MM/YYYY"
        format="DD/MM/YYYY"
        component={(props) => <input {...props} />}
      />
```

If you change the month and click outside the overlay, the overlay won't close if you click outside the overlay and stay opened

**To Reproduce**
Add a custom component as [the example in the doc](http://react-day-picker.js.org/api/DayPickerInput#component)

https://codesandbox.io/s/react-day-picker-base-ym4h5

**Expected behavior**
The same as without a custom component

**Desktop (please complete the following information):**

- OS: MacOS Cataline
- Browser : Chrome
- Version : 80.0.3977.0

**Comment by webunderconstruction** [comment]
If you focus back to the input, then you can click away to close the overlay, still a bug tho...

**Comment by john-rodewald** [comment]
Issue persists. Would appreciate any input on a fix.

**Comment by sibstark** [comment]
Hi! I faced this bug, will be any fixes for custom input?

**Comment by heyFahad** [comment]
The creator does not seem to be serious about fundamental issues in the current version of the package. @gpbl?

**Comment by medev21** [comment]
I'm also facing this issue, any word on a fix update?

**Comment by tjfred35** [comment]
I'd help here, but I can't reproduce on my version of chrome or firefox.

**Comment by heyFahad** [comment]
I finally figured out the solution. I used `react-overlay` to create my custom DateRangePicker.
